pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -161,6 +161,25 @@ func convertEndpoint(service *model.Service, servicePort *networking.Port,
 	}
 }
 
+// convertWorkloadInstances translates a WorkloadEntry into ServiceInstances. This logic is largely the
+// same as the ServiceEntry convertInstances.
+func convertWorkloadInstances(wle *networking.WorkloadEntry, services []*model.Service, se *networking.ServiceEntry) []*model.ServiceInstance {
+	out := make([]*model.ServiceInstance, 0)
+	for _, service := range services {
+		for _, port := range se.Ports {
+			ep := convertEndpoint(service, port, wle)
+			// WorkloadEntries always are using mTLS
+			ep.Endpoint.TLSMode = model.IstioMutualTLSModeLabel
+			if ep.Endpoint.Labels == nil {
+				ep.Endpoint.Labels = map[string]string{}
+			}
+			ep.Endpoint.Labels[model.TLSModeLabelName] = model.IstioMutualTLSModeLabel
```

Comment:
and the user could choose to explicitly add this label for certain cross cluster scenarios.

Reply:
So with the above SA change, we don't need to add this label, right? Since we are looking at the tlsMode option, not the label?